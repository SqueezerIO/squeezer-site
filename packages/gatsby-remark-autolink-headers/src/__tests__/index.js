const Remark = require(`remark`)
const visit = require(`unist-util-visit`)

const plugin = require(`../`)

const remark = new Remark().data(`settings`, {
  commonmark: true,
  footnotes: true,
  pedantic: true,
})

describe(`gatsby-remark-autolink-headers`, () => {
  it(`adds id to a markdown header`, () => {
    const markdownAST = remark.parse(`# Heading Uno`)

    const transformed = plugin({ markdownAST })

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()

      expect(node).toMatchSnapshot()
    })
  })

  it(`adds ids to each markdown header`, () => {
    const markdownAST = remark.parse(`
# Heading One

## Heading Two

### Heading Three
    `)

    const transformed = plugin({ markdownAST })

    visit(transformed, `heading`, node => {
      expect(node.data.id).toBeDefined()
    })
  })
})
