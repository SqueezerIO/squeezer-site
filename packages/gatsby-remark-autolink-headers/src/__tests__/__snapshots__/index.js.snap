// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-remark-autolink-headers adds id to a markdown header 1`] = `
Object {
  "children": Array [
    Object {
      "data": Object {
        "hChildren": Array [
          Object {
            "type": "raw",
            "value": "<svg aria-hidden=\\"true\\" height=\\"16\\" version=\\"1.1\\" viewBox=\\"0 0 16 16\\" width=\\"16\\"><path fill-rule=\\"evenodd\\" d=\\"M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z\\"></path></svg>",
          },
        ],
        "hProperties": Object {
          "aria-hidden": true,
          "class": "anchor",
        },
      },
      "title": null,
      "type": "link",
      "url": "#heading-uno",
    },
    Object {
      "position": Position {
        "end": Object {
          "column": 14,
          "line": 1,
          "offset": 13,
        },
        "indent": Array [],
        "start": Object {
          "column": 3,
          "line": 1,
          "offset": 2,
        },
      },
      "type": "text",
      "value": "Heading Uno",
    },
  ],
  "data": Object {
    "hProperties": Object {
      "id": "heading-uno",
    },
    "htmlAttributes": Object {
      "id": "heading-uno",
    },
    "id": "heading-uno",
  },
  "depth": 1,
  "position": Position {
    "end": Object {
      "column": 14,
      "line": 1,
      "offset": 13,
    },
    "indent": Array [],
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "heading",
}
`;
